swift/Workflow/Sources/RenderContext.swift

```diff
@@ -75,6 +75,10 @@ public class RenderContext<WorkflowType: Workflow>: RenderContextType {
         fatalError()
     }
 
+    public func runSideEffect(key: AnyHashable, action: (Lifetime) -> Void) {
```

Comment:
This needs documentation, especially discussing that the side effect is allowed to send actions to the context's `Sink`, that the side effect is expected to clean itself up when the `Lifetime` is ended, that the `Lifetime` won't end until a render pass in which this function is not called with the given key, etc.

Reply:
Also worth noting that the given closure will not be executed on subsequent runs of the same side-effect key.